coursier.bzl

```diff
@@ -268,7 +268,23 @@ def _generate_imports(repository_ctx, dep_tree, neverlink_artifacts, override_ta
             if (neverlink_artifacts.get(_strip_packaging_and_classifier_and_version(artifact["coord"]))):
                 target_import_string.append("\tneverlink = True,")
 
-            # 7. Finish the java_import rule.
+            # 7. If `strict_visibility` is True in the artifact spec, define public
+            #    visibility only for non-transitive dependencies.
+            #
+            # java_import(
+            # 	name = "org_hamcrest_hamcrest_library",
+            # 	jars = ["https/repo1.maven.org/maven2/org/hamcrest/hamcrest-library/1.3/hamcrest-library-1.3.jar"],
+            # 	srcjar = "https/repo1.maven.org/maven2/org/hamcrest/hamcrest-library/1.3/hamcrest-library-1.3-sources.jar",
+            # 	deps = [
+            # 		":org_hamcrest_hamcrest_core",
+            # 	],
+            #   tags = ["maven_coordinates=org.hamcrest:hamcrest.library:1.3"],
+            #   neverlink = True,
+            #   visibility = ["//visibility:public"],
+            if (repository_ctx.attr.strict_visibility and explicit_artifacts.get(_strip_packaging_and_classifier_and_version(artifact["coord"]))):
```

Comment:
```suggestion
            if repository_ctx.attr.strict_visibility and explicit_artifacts.get(target_label):
```

Reply:
Oh, I made a mistake here. target_label is escaped, but you're comparing with the non-escaped version. Feel free to revert.

Reply:
Ha, yeah, _just_ ran into that. Backing out.